### PR TITLE
Adds the task request modal to task details page

### PR DESCRIPTION
**Pull request by Ajeyakrishna-k**
### Issue:
- https://github.com/Real-Dev-Squad/website-status/issues/963

### Description:
- Adds task request modal to task details page. Task requests created here will be of the type assignment and on approval the task will be assigned to the requestor.
- Updates the RTK query for the new api endpoint.
- Updates the tests

### Dev Tested:
- [x] Yes

### Images/video of the change:
Screenshot 2023-11-17 at 8 40 08 PM

Screenshot 2023-11-17 at 8 45 01 PM

Screenshot 2023-11-17 at 8 44 36 PM

### CODE REVIEW

File: __tests__/Unit/Components/Tasks/TaskDetails.test.tsx [hunk 1]
```diff
@@ -623,11 +622,12 @@ describe('Task Details > Task Request', () => {
...
 
         await waitFor(() => {
-            screen.getByText(/successfully requested for task/i);
+            const taskRequestModalTitle = screen.getByText(/Task Request/i);
+            expect(taskRequestModalTitle).toBeInTheDocument();
         });
     });
 
-    it('Error toast should be shown on error', async () => {
+    test('Error toast should be shown on error', async () => {
```
Comment (ajoykumardas12) [inline __tests__/Unit/Components/Tasks/TaskDetails.test.tsx:630]: (Not a change request) when should we use test and when it?

File: src/components/taskDetails/index.tsx [hunk 2]
```diff
@@ -469,10 +486,19 @@ const TaskDetails: FC<Props> = ({ taskID }) => {
                                         <button
                                             data-testid="request-task-button"
                                             className={classNames.button}
-                                            onClick={taskRequestHandle}
+                                            onClick={toggle}
```
[the remark this answers is not in the document]
Comment (Ajeyakrishna-k) [reply src/components/taskDetails/index.tsx]: Done.

File: __tests__/Unit/Components/Tasks/TaskDetails.test.tsx (hunk 1 above)
Comment (Ajeyakrishna-k) [reply __tests__/Unit/Components/Tasks/TaskDetails.test.tsx]: I think they both behave the same. Anyways changed it back to it

**Review by ajoykumardas12 — APPROVED** [review]

File: src/components/issues/TaskRequestForm.tsx [hunk 3]
```diff
@@ -100,7 +100,7 @@ const TaskRequestForm: FC<ActionFormProps> = ({
     const handleSubmit = async (e: MouseEvent) => {
         e.preventDefault();
         setIsLoading(true);
-        await createTaskRequest(state);
+        if (createTaskRequest) await createTaskRequest(state);
```
Comment (shubhamsigdar1) [inline src/components/issues/TaskRequestForm.tsx:103]: Nit: How createTaskRequest here is working??

File: src/components/issues/ActionForm.tsx [hunk 4]
```diff
@@ -108,15 +108,15 @@ const ActionForm: FC<ActionFormProps> = ({
...
+        if (createTask) await createTask(state);
         setIsLoading(false);
     };
 
     const handleUpdateTask = async (e: MouseEvent) => {
         e.preventDefault();
         setIsLoading(true);
         setShowSuggestion(false);
-        await updateTask(state, taskId);
+        if (updateTask) await updateTask(state, taskId);
```
Comment (shubhamsigdar1) [inline src/components/issues/ActionForm.tsx:119]: When is updateTask function worked here??

File: src/components/taskDetails/index.tsx [hunk 5]
```diff
@@ -174,15 +182,24 @@ const TaskDetails: FC<Props> = ({ taskID }) => {
         }));
     }
 
-    function taskRequestHandle() {
-        if (!userData) {
-            return;
+    const handleCreateTaskRequest = async (data: TaskRequestData) => {
```
[the remark this answers is not in the document]
Comment (Ajeyakrishna-k) [reply src/components/taskDetails/index.tsx:185]: Caller passes the arguments? 
The screenshot that you posted is not the place where this function is called.

File: src/components/issues/TaskRequestForm.tsx [hunk 6]
```diff
@@ -15,7 +15,7 @@ type ActionFormReducer = {
 type ActionFormProps = {
     requestId?: string;
     taskId?: string;
-    createTaskRequest: (data: ActionFormReducer) => Promise<void>;
+    createTaskRequest?: (data: ActionFormReducer) => Promise<void>;
```
[the remark this answers is not in the document]
Comment (Ajeyakrishna-k) [reply src/components/issues/TaskRequestForm.tsx:18]: Data type of the form reducer.

File: src/components/issues/TaskRequestForm.tsx (hunk 3 above)
Comment (Ajeyakrishna-k) [reply src/components/issues/TaskRequestForm.tsx:103]: Gets passed in props

File: src/components/issues/ActionForm.tsx (hunk 4 above)
Comment (Ajeyakrishna-k) [reply src/components/issues/ActionForm.tsx:119]: It calls the update task function passed from the parent element.

**Review by shubhamsigdar1 — APPROVED** [review]

**Review by sahsisunny — APPROVED** [review]

**Pull request merged**